android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -407,8 +412,17 @@ constructor(
       )
       return
     }
-
-    defaultRepository.addOrUpdate(resource = questionnaireResponse)
+    defaultRepository.addOrUpdate(resource = questionnaireResponse) // update resource
+    // update the group/Or any other resource which is associated with this resource
+
+    if (paramDataActionParameter.isNotEmpty() && paramDataActionParameter[0].value.contains("/")) {
```

Comment:
If we use a distinct `ActionParameterType`, we will not need this check plus this assumption is not totally correct. We might have other parameter types that contain `/` that is not of our interest.